docs/topics/object-declarations.md

```diff
@@ -193,32 +193,91 @@ fun main() {
 }
 ```
 
-Just like [data classes](data-classes.md), you can mark your `object` declaration with the `data` modifier to get a nicely formatted string representation without having to manually provide an implementation for its `toString` function:
+Just like [data classes](data-classes.md), you can mark an `object` declaration with the `data` modifier. This instructs the compiler to generate a number of functions for your object:
 
+* `toString()` returning the name of the data object
+* `equals()`/`hashCode()` pair
+
+Note that you can't provide a custom `equals` or `hashCode` implementation for a `data object`.
+
+The `toString()` function of a data object returns the name of the object:
 ```kotlin
-data object MyObject
+data object MyDataObject {
+    val x: Int = 3
+}
 
 fun main() {
-    println(MyObject) // MyObject
+    println(MyDataObject) // MyDataObject
 }
 ```
 
-[Sealed class hierarchies](sealed-classes.md) are a particularly good fit for `data object` declarations, since they allow you to maintain symmetry with any data classes you might have defined alongside the object:
+The `equals()` function for a `data object` ensures that all objects that have the type of your `data object` are considered equal.
+In most cases, you will only have a single instance of your data object at runtime (after all, a `data object` declares a singleton).
```

Comment:
"The `hashCode()` function is implemented to be consistent with `equals()`, so that all instances of a certain data object class will have the same hash code."

Reply:
I've expanded the paragraph below the code sample. Do you want the `hashCode` function mentioned here, instead?